loader/src/main/java/net/minecraftforge/fml/loading/moddiscovery/AbstractModProvider.java

```diff
@@ -38,7 +38,7 @@ protected IModLocator.ModFileOrException createMod(Path... path) {
         var sj = SecureJar.from(
                 Manifest::new,
                 jar -> jar.moduleDataProvider().findFile(MODS_TOML).isPresent() ? mjm : JarMetadata.from(jar, path),
-                (root, p) -> true,
+                null,
```

Comment:
I've tested this specific class' change in production using my Java agent and an equivalent ASM transformer. The other changes have not been tested yet.